src/resources/TeamContributionCalendar/TeamContributionCalendar.test.js

```diff
@@ -109,6 +118,26 @@ describe('TeamContributionCalendar', () => {
         calendarHeader,
       )).to.equal(true);
     });
+
+    it('gets the tooltip element', () => {
+      teamContributionCalendar.renderActualCalendar();
+
+      expect(tooltipStub.calledOnce).to.equal(true);
+    });
+
+    it('appends the tooltip element to the container', () => {
+      teamContributionCalendar.renderActualCalendar();
+
+      expect(calendarContainer.appendChild.calledWithExactly(
+        calendarTooltip,
+      )).to.equal(true);
+    });
+
+    it('adds the tooltip events to the `rect` elements', () => {
```

Comment:
Not what the test does exactly. This sounds like something that could be tested directly.

Reply:
According to our discussion above, would it be reasonable to remove this case as it doesn't test any behavior related stuff?

Reply:
`adds the tooltip events to the rect elements` => regardless of how the test is written now, this is something that tested on the elements, no?

Reply:
Yes, that's it.

But writing `calls addEventsToRectElements` wouldn't fit here either. The solution is to simply omit the case, isn't it?

Reply:
can't you test the rendered elements that they have the proper event on them?

Reply:
Not sure, let me take a look into it.

Reply:
`aggregateUserCalendars` does have side effects, it updates the calendar.

Reply:
As the testing of `addEventsToRectElements` is pretty complex without the "normal" `document` object, the related test case has been removed.